code/modules/surgery/robolimbs.dm

```diff
@@ -164,3 +164,49 @@
 	span_warning("Your hand slips, damaging connectors on [target]'s [affected.display_name]!"))
 	target.balloon_alert_to_viewers("Slipped!")
 	target.apply_damage(10, BRUTE, affected, 0, TRUE, updating_health = TRUE)
+
+/datum/surgery_step/limb/biotic/attach
```

Comment:
Just change limb/attach to accept extra tools and do a check during end_step for whether to robotize or not, everything else here is duplicate code.

Reply:
except you cant change var type and robot_perts is not a child of limb